scripts/release-candidate.sh

```diff
@@ -0,0 +1,115 @@
+        -a              automatic mode. Requires -v parameter to be 100% without prompt
+        -s              same as -a but in silent mode
+
+        For Example: ./release-candidate.sh -v 3.5.0 -b candidate -a
+
+        -h              Help
+"
+}
+NO_PROMPT=0
+
+while getopts ":v:b:dhas" opt; do
+  case $opt in
+    v) VERSION="$OPTARG"
+    ;;
+    b) UPSTREAM_BRANCH="$OPTARG"
+    ;;
+    d) set -ex
+    ;;
+    h) show_help; exit 0
+    ;;
+    a) NO_PROMPT=1
+    ;;
+    s) NO_PROMPT=1; exec &>/dev/null
+    ;;
+    \?) echo "Invalid option -$OPTARG" >&2; exit 1
+    ;;
+  esac
+done
+
+if [ -z "$UPSTREAM_BRANCH" ]
+  then
+    UPSTREAM_BRANCH=candidate
+fi
+
+if [ -z "$VERSION" ]
+  then
+    read -r -p "Please input the version number to release (ex: for v3.5.0 please input 3.5.0): " VERSION
+fi
+
+UPSTREAM=git@github.com:rakutentech/ios-miniapp.git
+CANDIDATE_BRANCH=$VERSION/$UPSTREAM_BRANCH
+WORK_BRANCH=$(git branch --show-current)
+
+if [ $NO_PROMPT == 0 ]
+then
+  while true; do
+      echo "Before validating check every features and bug fixes are present on upstream master branch"
+      read -r -p "Is the next release v$VERSION version number correct? [Y/n] " yn
+      case $yn in
+          [Yy]* ) break;;
+          [Nn]* ) read -r -p "Please input the version number again (ex: for v3.5.0 please input 3.5.0): " VERSION;;
+          * ) break;;
+      esac
+  done
+fi
+
+echo "Stashing your WIP and creating local $UPSTREAM_BRANCH branch from upstream master branch"
+git stash
+git fetch "$UPSTREAM" master:"$CANDIDATE_BRANCH"
+git checkout "$CANDIDATE_BRANCH"
+
+echo "Retrieving current version number..."
+CURRENT_VERSION=$(grep -o -m 1 -E "([0-9]{1,}\.)+([0-9]{1,}\.)+[0-9]{1,}" MiniApp.podspec)
+SEARCH_STRING=$(grep "s.version   " MiniApp.podspec)
+echo "MiniApp.podspec mentions this at version line: $SEARCH_STRING"
+
+if [ $NO_PROMPT == 0 ]
+then
+  while true; do
+      read -r -p "Is v$CURRENT_VERSION the current version to upgrade. Input n to change it. [Y/n] " answer
+      case $answer in
+          [Yy]* ) break;;
+          [Nn]* ) read -r -p "Please input the s.version value currently in MiniApp.podspec (ex: for v3.5.0 please input 3.5.0): " CURRENT_VERSION;;
+          * ) break;;
+      esac
+  done
+fi
+
+REPLACE_STRING="  s.version      = '$VERSION'"
+
+echo "Updating MiniApp.podspec file s.version variable with v$VERSION version number"
+sed -i "" -e "s/$SEARCH_STRING/$REPLACE_STRING/" MiniApp.podspec
+
+echo "Changelog:"
+awk -v version="$VERSION" '/### / {printit = $2 == version}; printit;' CHANGELOG.md
+
+if [ $NO_PROMPT == 0 ]
+then
+  read -r -p "Is CHANGELOG.md up to date? Input Y after you think it is, or N to abort release[Y/n]: " answer
+
+  if [ "$answer" != "" ] && [ "$answer" == "${answer#[Nn]}" ] ;then
+      echo "Can't continue execution. Proceed manually"
+      exit 1
+  fi
+fi
+
+echo "Committing candidate version number changes"
+git commit -a -m "v$VERSION candidate"
```

Comment:
This is actually updating the version for the release, not just the candidate. So maybe the commit message should be something like `build: Update podspec to v$VERSION`?

Also, it seems a little dangerous to use `-a`. Could you instead just use `git add MiniApp.podspec`?

Reply:
absolutely